src/BugsnagUnity.mm

```diff
@@ -168,23 +168,45 @@
 
 }
 
-void bugsnag_getEventMetaData(const void *event,const void *instance, const char *tab, void (*callback)(const void *instance, const char *keys[], int keys_size, const char *values[], int values_size)) {
-     NSDictionary* sectionDictionary = [((__bridge BugsnagEvent *)event) getMetadataFromSection:@(tab)];
-     NSArray *keys = [sectionDictionary allKeys];
-     NSArray *values = [sectionDictionary allValues];
-     int count = 0;
-     if ([keys count] <= INT_MAX) {
-       count = (int)[keys count];
-     }
-     const char **c_keys = (const char **) malloc(sizeof(char *) * ((size_t)count + 1));
-     const char **c_values = (const char **) malloc(sizeof(char *) * ((size_t)count + 1));
-     for (NSUInteger i = 0; i < (NSUInteger)count; i++) {
-       c_keys[i] = [[keys objectAtIndex: i] UTF8String];
-       c_values[i] = [[[values objectAtIndex: i]description] UTF8String];
-     }
-    callback(instance,c_keys,count,c_values,count);
-    free(c_keys);
-    free(c_values);
+const char * getMetadataJson(NSDictionary* dictionary){
```

Comment:
Doing a nil check here would avoid having to do it in `bugsnag_getEventMetaData`, `bugsnag_getBreadcrumbMetadata` and `bugsnag_retrieveMetaData` - resulting in simpler code.

```suggestion
const char * getMetadataJson(NSDictionary* dictionary){
    if (!dictionary) {
        return NULL;
    }
```

Reply:
Done 👍